pkgs/by-name/gp/gprename/package.nix

```
@@ -0,0 +1,78 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  makeWrapper,
+  moreutils,
+  perlPackages,
+  gettext,
+  glib,
+  gtk3,
+  gobject-introspection,
+  pango,
+  harfbuzz,
+  gdk-pixbuf,
+  at-spi2-atk,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gprename";
+  version = "20230429";
+
+  src = fetchzip {
+    url = "mirror://sourceforge/projects/gprename/files/gprename/${finalAttrs.version}/gprename-${finalAttrs.version}.zip";
+    hash = "sha256-Du9OO2qeB1jUEJFcVYmLbJAGi2p/IVe3sqladq09AyY=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    moreutils
+  ];
+
+  postPatch = ''
+    grep -Ev 'desktop-file-install|update-desktop-database' Makefile | sponge Makefile
+
+    substituteInPlace Makefile \
+      --replace '/usr/share' '$(DESTDIR)/share'
+
+    substituteInPlace bin/gprename \
+      --replace '/usr/share' $out/share \
+      --replace '/usr/local/share' $out/share
+  '';
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+  buildInputs = [
+    perlPackages.perl
+    pango
+  ];
+  postInstall = ''
+    wrapProgram $out/bin/gprename \
+      --set PERL5LIB ${
+        perlPackages.makeFullPerlPath (
+          with perlPackages; [
+            Pango
+            Glib
+            Gtk3
+            LocaleGettext
+            libintl-perl
+          ]
+        )
+      } \
+      --prefix GI_TYPELIB_PATH : ${
+        lib.makeSearchPath "/lib/girepository-1.0" [
+          gtk3
+          pango.out
+          harfbuzz
+          gdk-pixbuf
+          at-spi2-atk
+        ]
+      }
```

```suggestion
      --set PERL5LIB ${
        perlPackages.makeFullPerlPath (with perlPackages; [ Pango Glib Gtk3 LocaleGettext libintl-perl ])} \
      --prefix GI_TYPELIB_PATH : ${lib.makeSearchPath "/lib/girepository-1.0" [ gtk3 pango.out harfbuzz gdk-pixbuf at-spi2-atk ]}
```
we can write that a lot more compact

---

We could, but when adding dependencies in future versions or if I missed something,
the diff would be less readable.
While there is still discussion in RFC 166 on when lists should be expanded, I think 5 elements is more than enough.

---

I like to put such lists on `let` blocks so that the bash code looks more readable.

But it's my personal nitpick - and I am not very interested on pursuing such [beautification wars](https://github.com/NixOS/nixpkgs/pull/278142#issuecomment-1873501290).